### PR TITLE
feat: styles plugins

### PR DESCRIPTION
**Pull request by supproduction**
Adds style plugins.
Changes service\builder for returning observables.

closes [HRZ-90296](https://spryker.atlassian.net/browse/HRZ-90296)

[HRZ-90296]: https://spryker.atlassian.net/browse/HRZ-90296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/platform/experience/src/services/layout/index.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export * from './default-layout.builder';
 export * from './default-layout.service';
 export * from './default-screen.service';
+export * from './layout-styles.utilities';
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/index.ts:4]: seems we're missing plugins here

Comment (supproduction) [reply libs/platform/experience/src/services/layout/index.ts:4]: we have it on line 8

File: libs/platform/experience/src/mixins/layout.mixin.ts [hunk 2]
```diff
@@ -159,37 +162,51 @@ export const LayoutMixin = <T extends Type<LitElement & LayoutAttributes>>(
...
-        this.attributeFilter,
+        featureVersion >= '1.2'
+          ? this.attributeFilter
+          : ['layout', ...layoutKeys],
         this.$options().rules
       )
     );
 
-    protected screen = computed(
+    protected screen$ = computed(
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/mixins/layout.mixin.ts]:
```suggestion
    protected $screen = computed(
```

File: libs/platform/experience/src/services/layout/plugins/properties/sticky/sticky-layout.plugin.ts [hunk 3]
```diff
@@ -13,15 +13,17 @@ export class StickyLayoutPlugin implements LayoutPlugin {
...
 
-  getStyleProperties(data: LayoutStyleProperties): LayoutPluginStyleProperties {
-    return {
+  getStyleProperties(
+    data: LayoutStyleParameters
+  ): Observable<LayoutStyleProperties> {
+    return of({
       'max-height': `calc(${data.height ?? '100vh'} - ${data.top ?? '0px'})`,
-    };
+    });
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/properties/sticky/sticky-layout.plugin.ts]: I thought we're going to have the ability to resolve the schema properties from this plugin as well as from other plugins.

File: libs/platform/experience/src/services/layout/plugins/styles/typography/typography-layout.plugin.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    data: LayoutStyleParameters
+  ): Observable<LayoutStyleProperties> {
+    if (!data.typography) return of({});
+
+    return of({
+      'font-size': `var(--oryx-typography-${data.typography}-size)`,
+      'font-weight': `var(--oryx-typography-${data.typography}-weight)`,
+      'line-height': `var(--oryx-typography-${data.typography}-line)`,
+    });
+  }
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/styles/typography/typography-layout.plugin.ts]: The typography related model would ideally move out of the `composition.model.ts` and find home here in the model/schema.

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/platform/experience/src/services/layout/plugins/styles/misc/misc-layout.model.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+     */
+    gap?: string | number;
+
+    /**
+     * Specifies the stack order of a component. This is useful in combination with sticky layouts,
+     * where the stacking order cannot be controlled by static css rules.
+     *
+     * Defaults to `1` in sticky layouts.
+     */
+    zIndex?: number;
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/styles/misc/misc-layout.model.ts]: let's move it to sticky plugin.

File: libs/platform/experience/src/services/layout/plugins/styles/misc/misc-layout.model.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+     * or 1 columns for large, medium and small screens.
+     *
+     * While the column system can be configured globally by design tokens, the `columnCount` allows
+     * for an individual override per layout.
+     *
+     * It is also worth mentioning that nested layouts will inherit the column size from the parent
+     * layout, by taken the parent item _span_ count, which default to `1`. This results in a
+     * balanced layout, where nested layouts will reflect the parent column setup.
+     */
+    columnCount?: number;
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/styles/misc/misc-layout.model.ts]: `columnCount`, `align`, `justify`, `span`, `colSpan`, `rowSpan`, `gridRow`, `gridColumn` and `gap` are all related to layout. Used by many layouts. Could we move them to a layout style plugin? Or can we use them in different layout plugins (e.g. column, carousel and grid)? In the latter case, they would be used multiple times.

File: libs/platform/experience/src/services/layout/plugins/styles/misc/misc-layout.model.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    /**
+     * Very useful for stick layouts that need to allow for scrollable overflow, typically
+     * in combination with sticky layout. The side navigation aside the page content is a
+     * good example where this is needed, since the aside navigation is often setup to stick
+     * to the top so that it won't scroll away.
+     *
+     * The overflow cannot be hardcoded for sticky layouts, since there are sticky layouts that
+     * require explicit overflows (eg. typeahead search inside a sticky header)
+     */
+    overflow?: string;
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/styles/misc/misc-layout.model.ts]: let's move to sticky

File: libs/platform/experience/src/services/layout/plugins/styles/misc/misc-layout.model.ts [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+    /**
+     * Specifies the background of the item. The background can be used to set a color or even
+     * a background image.
+     */
+    background?: string;
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/styles/misc/misc-layout.model.ts]: let's move background, fill, border and radius to `canvas` styles plugin. We might come with a better name over time. We're on our way to eliminate `misc`.

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/platform/experience/src/services/layout/plugins/styles/canvas/canvas-style.model.ts [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+     * or 1 columns for large, medium and small screens.
+     *
+     * While the column system can be configured globally by design tokens, the `columnCount` allows
+     * for an individual override per layout.
+     *
+     * It is also worth mentioning that nested layouts will inherit the column size from the parent
+     * layout, by taken the parent item _span_ count, which default to `1`. This results in a
+     * balanced layout, where nested layouts will reflect the parent column setup.
+     */
+    columnCount?: number;
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/styles/canvas/canvas-style.model.ts]: layout related styles are not expected here

File: libs/platform/experience/src/services/layout/layout.model.ts [hunk 10]
```diff
@@ -9,6 +9,21 @@ declare global {
   export interface LayoutsProperty {
     vertical?: boolean;
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/layout.model.ts]: This one could move to LayoutStyles

File: libs/platform/experience/src/services/layout/layout.model.ts [hunk 11]
```diff
@@ -9,6 +9,21 @@ declare global {
   export interface LayoutsProperty {
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/layout.model.ts]: Should we name it `LayoutProperty`?

File: libs/platform/experience/src/services/layout/plugins/properties/divider/divider-layout.plugin.ts [hunk 12]
```diff
@@ -8,9 +8,9 @@ export class DividerLayoutPlugin implements LayoutPlugin {
     return ssrAwaiter(import('./divider.styles').then((m) => m.styles));
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/properties/divider/divider-layout.plugin.ts:8]: What about renaming the styles `dividerStyles` and export the styles in `index.ts`?

Comment (tobi-or-not-tobi) [reply libs/platform/experience/src/services/layout/plugins/properties/divider/divider-layout.plugin.ts:8]: if we do this, we should do it everywhere

File: libs/platform/experience/src/services/layout/plugins/styles/canvas/canvas-style.schema.ts [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+const alignOptions = [
+  { value: LayoutAlign.Start },
+  { value: LayoutAlign.End },
+  { value: LayoutAlign.Stretch },
+  { value: LayoutAlign.Center },
+  { value: LayoutAlign.SpaceBetween },
+  { value: LayoutAlign.SpaceAround },
+  { value: LayoutAlign.SpaceEvenly },
+];
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/styles/canvas/canvas-style.schema.ts]: not in use

File: libs/platform/experience/src/services/layout/plugins/styles/canvas/canvas-style.model.ts [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+     * Sets an element's border, using a width, style and color (e.g. `1px solid red`)
+     */
+    border?: string;
+
+    /**
+     * Rounds the corners of an element's outer border edge.
+     */
+    radius?: string | number;
+
+    ratio?: string;
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/services/layout/plugins/styles/canvas/canvas-style.model.ts]: we have this one also at spacing model

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/platform/experience/src/services/layout/layout.model.ts (hunk 10 above)
Comment (tobi-or-not-tobi) [reply libs/platform/experience/src/services/layout/layout.model.ts]: Where do we render the vertical property?

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**